packages/wallets/src/features/cfd/screens/MT5TradeScreen/MT5TradeLink/MT5TradeLink.tsx

```diff
@@ -1,47 +1,121 @@
+import DerivXLabelIcon from '../../../../../public/images/derivx-label.svg';
+import LinuxIcon from '../../../../../public/images/ic-linux-logo.svg';
+import MacOSIcon from '../../../../../public/images/ic-macos-logo.svg';
 import MT5Icon from '../../../../../public/images/ic-mt5.svg';
 import WindowsIcon from '../../../../../public/images/ic-windows-logo.svg';
-import MacOSIcon from '../../../../../public/images/ic-macos-logo.svg';
-import LinuxIcon from '../../../../../public/images/ic-linux-logo.svg';
+import { THooks, TPlatforms } from '../../../../../types';
+import { PlatformToTitleMapper } from '../../../constants';
 import './MT5TradeLink.scss';
 
 const AppToContentMapper = {
+    ctrader: {
+        icon: <WindowsIcon />,
+        link: 'https://getctrader.com/deriv/ctrader-deriv-setup.exe',
+        text: 'Download',
+        title: 'CTrader Windows App',
+    },
     linux: {
         icon: <LinuxIcon />,
+        link: 'https://www.metatrader5.com/en/terminal/help/start_advanced/install_linux',
         text: 'Learn more',
         title: 'MetaTrader 5 Linux app',
     },
     macos: {
         icon: <MacOSIcon />,
+        link: 'https://download.mql5.com/cdn/web/metaquotes.software.corp/mt5/MetaTrader5.dmg',
         text: 'Download',
         title: 'MetaTrader 5 MacOS app',
     },
     web: {
         icon: <MT5Icon />,
+        link: '',
         text: 'Open',
         title: 'MetaTrader 5 web',
     },
     windows: {
         icon: <WindowsIcon />,
+        link: 'https://download.mql5.com/cdn/web/deriv.com.limited/mt5/deriv5setup.exe',
         text: 'Download',
         title: 'MetaTrader 5 Windows app',
     },
 };
 
+const PlatformToLabelIconMapper = {
+    ctrader: <CTraderLabelIcon />,
+    derivez: <DerivEZLabelIcon />,
+    dxtrade: <DerivXLabelIcon />,
+};
+
 type TMT5TradeLinkProps = {
-    app: keyof typeof AppToContentMapper;
+    app?: keyof typeof AppToContentMapper;
+    isDemo?: THooks.ActiveWalletAccount['is_virtual'];
+    platform?: TPlatforms.All;
+    webtraderUrl?: THooks.MT5AccountsList['webtrader_url'];
 };
 
-const MT5TradeLink: FC<TMT5TradeLinkProps> = ({ app }) => {
+const MT5TradeLink: FC<TMT5TradeLinkProps> = ({ app = 'linux', isDemo = false, platform, webtraderUrl = '' }) => {
     const content = AppToContentMapper[app];
+    const { data: ctraderToken } = useCtraderServiceToken();
+    const { data: derivezToken } = useDerivezServiceToken();
+
+    const onClickWebTerminal = () => {
+        const { isStaging, isTestLink } = getPlatformFromUrl();
+        let url;
+        switch (platform) {
+            case 'dxtrade':
+                url = isDemo ? 'https://dx-demo.deriv.com' : 'https://dx.deriv.com';
+                break;
+            case 'ctrader':
+                url = isTestLink || isStaging ? 'https://ct-uat.deriv.com' : 'https://ct.deriv.com';
+                if (ctraderToken) url += `?token=${ctraderToken}`;
```

Comment:
@markwylde-deriv do we have any better patterns than passing token in URL already, or is it not there yet?

Reply:
Am I right in thinking this is how we hand off to ctrader and mt5? If so, can ctrader not go to our oauth server directly and get it's own token? @yashim-deriv @balakrishna-deriv

Reply:
We have confirmation the token is a one time token, that expired immediately once used.